cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -224,11 +241,18 @@ func (c *Client) AddMember(name string, peerAddrs string) ([]Member, error) {
 	defer cli.Close()
 
 	// Adds a new member to the cluster
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	resp, err := cli.MemberAdd(ctx, []string{peerAddrs})
-	cancel()
+	var lastError error
+	var resp *clientv3.MemberAddResponse
+	err = wait.ExponentialBackoff(addRemoveBackoff, func() (bool, error) {
+		resp, err = cli.MemberAdd(context.Background(), []string{peerAddrs})
```

Comment:
please note that etcd does seem to have a number of different "retry clients", but those seem undocumented or at least i couldn't find the docs. i did not investigate further as this solution:
- gives us fine grained control over the backoff
- worked in my tests